test/async_utils_test.py

```diff
@@ -382,3 +384,71 @@ async def gen2():
         result.append(item)
 
     assert result == [(1, 2)]
+
+
+@pytest.mark.asyncio
+async def test_async_merge():
+    result = []
+    states = []
+
+    async def gen1():
+        states.append("gen1 enter")
+        try:
+            await asyncio.sleep(0.1)
```

Comment:
There is a small risk that these sleeps will flake out in CI where things run *sloooow* - not sure there is an easy way to implement the interleaving of generators using Events/Locks instead?

Reply:
changed to events & deterministic output order!